frontend/src/util/auth.tsx

```diff
@@ -77,9 +82,82 @@ class SimpleAuth {
 	}
 }
 
+class PasswordAuth {
+	currentUser: User | null = null
+	googleProvider?: Firebase.auth.GoogleAuthProvider
+	githubProvider?: Firebase.auth.GithubAuthProvider
+
+	constructor() {
+		const token = sessionStorage.getItem('passwordToken')
+		if (token) {
+			this.currentUser = fakeFirebaseUser as any
+		}
+	}
+
+	async createUserWithEmailAndPassword(
+		email: string,
+		password: string,
+	): Promise<Firebase.auth.UserCredential> {
+		return await getFakeFirebaseCredentials()
+	}
+
+	onAuthStateChanged(
+		onSignedIn: (user: Firebase.User | null) => void,
+		onError: (error: Firebase.auth.Error) => any,
+	): () => void {
+		onSignedIn(this.currentUser as Firebase.User)
+		return function () {}
+	}
+
+	sendPasswordResetEmail(email: string): Promise<void> {
+		return Promise.resolve(undefined)
+	}
+
+	async signInWithEmailAndPassword(
+		email: string,
+		password: string,
+	): Promise<Firebase.auth.UserCredential> {
+		const response = await fetch(`${PUBLIC_GRAPH_URI}/login`, {
+			method: 'POST',
+			body: JSON.stringify({
+				email,
+				password,
+			}),
+			headers: {
+				Accept: 'application/json',
+				'Content-Type': 'application/json',
+			},
+		})
+
+		if (response.status === 200) {
+			const jsonResponse = await response.json()
+			sessionStorage.setItem('passwordToken', jsonResponse.token)
+
+			this.currentUser = fakeFirebaseUser as any
+
+			const authResponse = await getFakeFirebaseCredentials()
+			return authResponse
+		}
+		throw new Error('Invalid credentials')
+	}
+
+	async signInWithPopup(
+		provider: Firebase.auth.AuthProvider,
+	): Promise<Firebase.auth.UserCredential> {
+		return await getFakeFirebaseCredentials()
+	}
+
+	signOut(): Promise<void> {
+		sessionStorage.removeItem('passwordToken')
+		return Promise.resolve(undefined)
+	}
+}
+
 export let auth: SimpleAuth
 if (import.meta.env.REACT_APP_AUTH_MODE === 'simple') {
 	auth = new SimpleAuth()
+} else if (import.meta.env.REACT_APP_AUTH_MODE === 'password') {
```

Comment:
could this use the new exported constant from `constants.ts`?

Reply:
Done